### PR TITLE
Add custom wire support for differentiable Hamiltonians

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
The `molecular_hamiltonian` function is modified to support custom wires for the differentiable Hamiltonian.  

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/qchem/openfermion_obs.py [hunk 1]
```diff
@@ -935,13 +939,17 @@ def molecular_hamiltonian(
         )
         if args is None:
             h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)()
+            if wires:
+                h = qml.map_wires(h, wires_map)
```
Comment (obliviateandsurrender) [inline pennylane/qchem/openfermion_obs.py:943]: 🍪

**Review by obliviateandsurrender — APPROVED** [review]
Thanks @soranjh, looks good to me!

**Review by Jaybsoni — APPROVED** [review]
Looks good! Minor comment, in tests we usually use `parameterize` when we have multiple sets of args we want to test. Otherwise it suffices to just define the args in the function. 

My (non-blocking) recommendation would be to add one or two more cases to the list, maybe using non-trivial wire labels (eg. `wiremap = [0 , "z", 3, "ancilla"]`).

Cheers,

**Comment by soranjh** [comment]
> Looks good! Minor comment, in tests we usually use `parameterize` when we have multiple sets of args we want to test. Otherwise it suffices to just define the args in the function.
> 
> My (non-blocking) recommendation would be to add one or two more cases to the list, maybe using non-trivial wire labels (eg. `wiremap = [0 , "z", 3, "ancilla"]`).
> 
> Cheers,

Added more tests.

**Comment by soranjh** [comment]
[sc-38037]

**Pull request merged**